target/linux/ramips/dts/mt7621_tplink_er605-v2.dts

```diff
@@ -12,6 +12,9 @@
 
 	chosen {
 		bootargs = "console=ttyS0,115200 noinitrd";
+		// Override bootargs because u-boot passes wrong root parameter.
+		// Instead allow kernel determine root automatically by looking for rootfs volume
+		bootargs-override = "console=ttyS0,115200 noinitrd ubi.mtd=3,2048";
```

Comment:
What does the 2048 in `ubi.mtd=3,2048"` do?
Sorry I am not very familiar with this part of the kernel.

Reply:
Stock FW has this, as well as u-boot.
I think this the ubi header offset on the device. http://www.linux-mtd.infradead.org/faq/ubi.html has some information on this.